### PR TITLE
Add sig-cluster-lifecycle-cluster-api-alerts google group via SIG Cluster Lifecycle delegated groups management.

### LINKED ISSUE #1863: Request for sig-cluster-lifecycle-cluster-api-alerts@kubernetes.io Google Group
We were using kubernetes-sig-cluster-lifecycle-cluster-api-alerts@googlegroups.com for Testgrid/Prow alerts for Cluster API. Unfortunately, after a Google Workspace migration, all the owners were removed and the group has disappeared into the ether.

Would it be possible to create a Google Group within the kubernetes.io domain, so we can manage it via this repository, e.g. in this example PR https://github.com/kubernetes/k8s.io/pull/1862 where I attempted to do it using a normal googlegroups.com email address?

Thanks.

### PR DESCRIPTION
**Pull request by randomvariable**
Fixes #1863 

Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

/assign @ameukam

### CODE REVIEW

File: groups/sig-cluster-lifecycle/groups.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    description: |-
+
+    settings:
+      WhoCanPostMessage: "ANYONE_CAN_POST"
+      ReconcileMembers: "true"
+    owners:
+      - detiber@gmail.com
+      - jdetiberus@equinix.com
+      - jeewan@vmware.com
+      - sbueringer@gmail.com
```
Comment (spiffxp) [inline groups/sig-cluster-lifecycle/groups.yaml:13]: I'll merge to allow us to iterate, but we need to stop making people owners of groups unless there are specific reasons to do so.

It's the same reason we disallow GitHub team owners.  If people can make manual changes, they will, and then get confused when our automation reverts their changes.

Comment (spiffxp) [reply groups/sig-cluster-lifecycle/groups.yaml:13]: Where would you want to see such a policy documented?  I plan on using tests to enforce, but that's the friendliest way to find out.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
But with caveats

File: groups/sig-cluster-lifecycle/groups.yaml (hunk 1 above)
Comment (ameukam) [reply groups/sig-cluster-lifecycle/groups.yaml:13]: 🤔 We can start with an example of how a mailing list should be created in https://github.com/kubernetes/k8s.io/tree/master/groups and also add a link to https://www.kubernetes.dev/docs/comms/mailing-lists/ (I just found out).

Comment (randomvariable) [reply groups/sig-cluster-lifecycle/groups.yaml:13]: Oh, I was just copying the community one that had some folk in OWNERS. Happy to move everyone to members

**Pull request merged**